WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/progress/BackupDownloadProgressFragment.kt

```diff
@@ -25,20 +35,37 @@ class BackupDownloadProgressFragment : Fragment(R.layout.backup_download_progres
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        initRecyclerView()
         initViewModel()
     }
 
     private fun initDagger() {
         (requireActivity().application as WordPress).component().inject(this)
     }
 
+    private fun initRecyclerView() {
+        recycler_view.layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
+        initAdapter()
+    }
+
+    private fun initAdapter() {
+        recycler_view.adapter = BackupDownloadProgressAdapter(imageManager, uiHelpers)
+    }
+
     private fun initViewModel() {
         parentViewModel = ViewModelProvider(requireActivity(), viewModelFactory)
                 .get(BackupDownloadViewModel::class.java)
 
         viewModel = ViewModelProvider(this, viewModelFactory)
                 .get(BackupDownloadProgressViewModel::class.java)
 
+        viewModel.uiState.observe(viewLifecycleOwner, { uiState ->
+            when (uiState) {
+                is Content -> showContent(uiState)
+                is Error -> ToastUtils.showToast(requireContext(), "Implement Error")
```

Comment:
Wrong import for `Error` here. `when` expression is not exhaustive because of it.

Reply:
I thought I caught all of those. Makes the case for using "Failure" instead of "Error". Thanks 🙇

Reply:
in 5d012c9